client/src/js/components/bhNavigation.js

```diff
@@ -18,64 +21,68 @@ function NavigationController($location, $rootScope, Tree, AppCache) {
    * parsed once - every following method should use the index to point to the
    * relevent unit
    */
-  var unitsIndex = {id : {}, path : {}};
+  var unitsIndex = { id : {}, path : {} };
 
   /** @todo handle exception cases displayed at the top of the Tree directive */
   Tree.units()
     .then(function (result) {
 
       Tree.sortByTranslationKey(result);
-      vm.units = result;
+      $ctrl.units = result;
 
-      calculateUnitIndex(vm.units);
-      expandInitialUnits(vm.units);
+      calculateUnitIndex($ctrl.units);
+      expandInitialUnits($ctrl.units);
     });
 
   // Tree Utility methods
-  vm.toggleUnit = function toggleUnit(unit) {
+  $ctrl.toggleUnit = function toggleUnit(unit) {
     unit.open = unit.open || false;
     unit.open = !unit.open;
 
     // Update cached record of modules expansion
     cache[unit.id] = { open : unit.open };
-    //cache.put(unit.id, {open : unit.open});
   };
 
-  vm.navigate = function navigate(unit) {
+  $ctrl.navigate = function navigate(unit) {
     selectUnit(unit);
     $location.path(unit.path);
   };
 
-  vm.refreshTranslation = function refreshTranslation() {
-    Tree.sortByTranslationKey(vm.units);
+  $ctrl.refreshTranslation = function refreshTranslation() {
+    Tree.sortByTranslationKey($ctrl.units);
   };
 
   /**
-   * Select a unit in the tree given aa specified URL
-   *
-   * @todo search through all of the path keys - if the recorded path is found
-   * /anywhere/ in the url the path should be selected in the tree
+   * Select a unit in the tree given a specified URL.
    */
-
   function trackPathChange(event, url) {
     var path = url.split('/#')[1];
-    var unit = unitsIndex.path[normalisePath(path)];
-
-    if (unit) {
-      selectUnit(unit);
-    }
+    var normPath = normalisePath(path);
+
+    /**
+     * loop through all paths, selecting those are match the selected url
+     *
+     * @todo - write test cases to be sure this works in all cases, probably
+     * dependent on the ordering of unitsIndex.
+     */
+    Object.keys(unitsIndex.path).forEach(function (key) {
+      var node = unitsIndex.path[key];
+      if (normPath.includes(node.path)) {
+        selectUnit(path);
+      }
+    });
   }
 
   function selectUnit(unit) {
 
     // Clear previous selection if it exists
-    if (vm.selectedUnit) {
-      vm.selectedUnit.selected = false;
+    if ($ctrl.selectedUnit) {
```

Comment:
The code does do what you suggest.  See lines 85 and 86, right below this check.  This `if` statement is trying to make sure the previous `unit.selected` is false.

Or am I misunderstanding?